src/main/java/edu/cornell/kfs/fp/document/web/struts/RecurringDisbursementVoucherForm.java

```diff
@@ -1,54 +1,105 @@
 import edu.cornell.kfs.fp.document.RecurringDisbursementVoucherDocument;
 import edu.cornell.kfs.fp.service.RecurringDisbursementVoucherDocumentService;
+import edu.cornell.kfs.fp.service.RecurringDisbursementVoucherPaymentMaintenanceService;
 
 public class RecurringDisbursementVoucherForm extends CuDisbursementVoucherForm {
-	
-	private static final long serialVersionUID = 7035540080454973823L;
-	private static final org.apache.log4j.Logger LOG = org.apache.log4j.Logger.getLogger(RecurringDisbursementVoucherForm.class);
-	
-	private transient List<RecurringDisbursementVoucherPDPStatus> pdpStatuses;
-	private transient RecurringDisbursementVoucherDocumentService recurringDisbursementVoucherDocumentService;
-	
-	@Override
-	public String getDocTypeName() {
-		return CuFPConstants.RecurringDisbursementVoucherDocumentConstants.RECURRING_DV_DOCUMENT_TYPE_NAME;
-	}
-	
-	@Override
+
+    private static final long serialVersionUID = 7035540080454973823L;
+    private static final org.apache.log4j.Logger LOG = org.apache.log4j.Logger.getLogger(RecurringDisbursementVoucherForm.class);
+
+    private transient List<RecurringDisbursementVoucherPDPStatus> pdpStatuses;
+    private transient RecurringDisbursementVoucherDocumentService recurringDisbursementVoucherDocumentService;
+    private transient RecurringDisbursementVoucherPaymentMaintenanceService recurringDisbursementVoucherPaymentMaintenanceService;
+
+    @Override
+    public String getDocTypeName() {
+        return CuFPConstants.RecurringDisbursementVoucherDocumentConstants.RECURRING_DV_DOCUMENT_TYPE_NAME;
+    }
+
+    @Override
     protected String getDefaultDocumentTypeName() {
         return CuFPConstants.RecurringDisbursementVoucherDocumentConstants.RECURRING_DV_DOCUMENT_TYPE_NAME;
     }
-	
-	@Override
-	public String getTripUrl() {
-		return StringUtils.EMPTY;
-	}
-	
-	@Override
-	public String getTripID() {
-		return StringUtils.EMPTY;
-	}
-	
-	public boolean isRecurringDVDetailsDefaultOpen() {
-		return !((RecurringDisbursementVoucherDocument)this.getDocument()).getRecurringDisbursementVoucherDetails().isEmpty();
-	}
-	
-	public int getRecurringDVDetailsSize() {
-	    return ((RecurringDisbursementVoucherDocument)this.getDocument()).getRecurringDisbursementVoucherDetails().size();
-	}
-	
-	public boolean isPreDisbursementProcessorTabDefaultOpen() {
-	    return !getPdpStatuses().isEmpty();
-	}
+
+    @Override
+    public String getTripUrl() {
+        return StringUtils.EMPTY;
+    }
+
+    @Override
+    public String getTripID() {
+        return StringUtils.EMPTY;
+    }
+
+    public boolean isCancelPDPPaymentsActionAvailable() {
+        if (getRecurringDisbursementVoucherPaymentMaintenanceService().hasCancelPermission(GlobalVariables.getUserSession().getPerson())) {
+            return arePaymentDetailsCancelAble() || areDVsCancelable();
+        }
+        return false;
+    }
+    
+    private boolean arePaymentDetailsCancelAble() {
```

Comment:
I think there is a misspelling in this method name: arePaymentDetailsCancelAble
I would also add 'Any' in the method name so change it to: areAnyPaymentDetailsCancelable
since not all details need be cancelable